addon/-private/system/model/internal-model.js

```diff
@@ -583,10 +583,19 @@ export default class InternalModel {
     let changedKeys;
 
     if (this.hasRecord) {
-      changedKeys = this._changedKeys(data.attributes);
+      if (typeof this._record._changedKeys === 'function') {
+        changedKeys = this._record._changedKeys(data.attributes);
+      } else {
+        changedKeys = this._changedKeys(data.attributes);
+      }
+    }
+
+    if (this.hasRecord && this._record._assignAttributes) {
```

Comment:
This could follow a similar pattern to what I mentioned RE: `_changedKeys`. We could add `this._assignAttributes` method and have it do the conditional?  Not sure that is useful though if this is literally the only place it happens...